.github/workflows/release.yml

```diff
@@ -11,79 +11,153 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
-        arch: [auto, aarch64]
+        arch: [native, aarch64, universal2]
         py: [cp37, cp38, cp39, cp310]
-        include:
-          - os: macos-latest
-            py: cp38
-            arch: universal2
-          - os: macos-latest
-            py: cp39
-            arch: universal2
-          - os: macos-latest
-            py: cp310
-            arch: universal2
         exclude:
           - os: windows-latest
             arch: aarch64
+          - os: windows-latest
+            arch: universal2
           - os: macos-latest
-            arch: aarch64
-    env:
-      CIBW_BUILD: ${{ matrix.py }}-*
-      CIBW_ARCHS_LINUX: ${{ matrix.arch }}
+            arch: aarch64 
+          - os: ubuntu-latest
+            arch: universal2
     steps:
-      - uses: actions/checkout@v2
+# Step for debugging    
+      - name: Show environment
+        run: env | grep ^GITHUB
+# Step for debugging
+      - name: Show CI env
+        run: env | grep ^CI  
+
+      - uses: actions/checkout@v3
         with:
           submodules: true
 
       - if: ${{ matrix.arch == 'aarch64' }}
-        uses: docker/setup-qemu-action@v1
+        uses: docker/setup-qemu-action@v2 
+
+      - uses: awvwgk/setup-fortran@main
+        id: setup-fortran
+        with:
+          compiler: gcc
+          version: 11
+
+# Step for debugging
+      - run: ${{ env.FC }} --version
+        env:
+          FC: ${{ steps.setup-fortran.outputs.fc }}     
+
+      - if: ${{ matrix.os == 'windows-latest' }}
+        uses: pypa/cibuildwheel@v2.9.0
+        env:
+          CIBW_BUILD_VERBOSITY: 3
+          CIBW_BUILD: ${{ matrix.py }}-*
+          CIBW_ARCHS: ${{ matrix.arch }}
+          CIBW_SKIP: "*musllinux*"
+          CMAKE_GENERATOR: "MinGW Makefiles"
+          FC: ${{ steps.setup-fortran.outputs.fc }}
+          CI: true
 
-      - uses: pypa/cibuildwheel@v2.3.1
+      - if: ${{ matrix.os != 'windows-latest' }}
+        uses: pypa/cibuildwheel@v2.9.0
         env:
+          # Check setuptools version
+          CIBW_BUILD_VERBOSITY: 3
           CIBW_BUILD: ${{ matrix.py }}-*
           CIBW_ARCHS: ${{ matrix.arch }}
+          CIBW_SKIP: "*musllinux*"
+          CIBW_TEST_SKIP: "*-macosx_universal2:arm64" 
+          CI: true  
 
-      - uses: actions/upload-artifact@v2
+      - uses: actions/upload-artifact@v3
         with:
-          path: ./wheelhouse/*.whl
+          path: ./wheelhouse/*.whl 
```

Comment:
I'd recommend using pre-commit and having it strip trailing whitespace. This change is completely pointless, it's just adding a trailing space.

Reply:
The pre-commit is installed, but currently it doesn't check formatting, only verify syntax. I am thinking to use `yamlfmt` from `https://github.com/jumanjihouse/pre-commit-hook-yamlfmt`.

Reply:
A hook to remove trailing whitespace is one of the standard hooks in pre-commit, see this excerpt from iminuit's version. I removed some standard hooks when I introduced pre-commit so that the diff wouldn't become too difficult to read. We should add `  - id: trailing-whitespace`
```
# Standard hooks
- repo: https://github.com/pre-commit/pre-commit-hooks
  rev: v4.1.0
  hooks:
  - id: check-case-conflict
  - id: check-docstring-first
  - id: check-executables-have-shebangs
  - id: check-merge-conflict
  - id: check-symlinks
  - id: check-yaml
    args: ["--allow-multiple-documents"]
  - id: debug-statements
  - id: end-of-file-fixer
  - id: mixed-line-ending
  - id: sort-simple-yaml
  - id: file-contents-sorter
  - id: trailing-whitespace
    exclude: ^doc/_static/.*.svg
```

Reply:
I see. I will add it.

Reply:
If you have external files, you can globally ignore them with pre-commit's outer configuration (`exclude:` at the top level, IIRC). Then pre-commit will never look at them or touch them on any check.

Reply:
It is also possible to use `files` to include files on which a hook could be run. So I've added:

```yaml
- id: trailing-whitespace
  files: '^\.github/workflows/|*\.py$'
```

It should run `trailing-whitespace` on all `.py` files and all files in `\.github/workflows`. I guess it is a safe choice.  Actually, the list of files in the project containing trailing whitespaces consists fortran files, cmake files, and couple of readme's. I prefer to do not touch them.